### DISCUSSION TITLE
Completion of error handling

### DISCUSSION THREAD

**Discussion by elfring** [pawelsalawa/sqlitestudio#4648] [General]
Would you like to add more error handling for return values from functions like the following?
* [fseek](https://pubs.opengroup.org/onlinepubs/9699919799/functions/fseek.html "Reposition a file-position indicator") ⇒ [Parse](https://github.com/pawelsalawa/sqlitestudio/blob/d5cd0d1d25bb872286b64da241143f536f5470ff/SQLiteStudio3/coreSQLiteStudio/parser/lemon.c#L2661)
* [malloc](https://pubs.opengroup.org/onlinepubs/9699919799/functions/malloc.html "Memory allocation") ⇒ [ParseCopyParserState](https://github.com/pawelsalawa/sqlitestudio/blob/d5cd0d1d25bb872286b64da241143f536f5470ff/SQLiteStudio3/coreSQLiteStudio/parser/lempar.c#L199)

**Comment by pawelsalawa** [comment]
Files that you're referring to are imported from official SQLite sources (though from quite an old version of SQLite). I'm not familiar enough with their details of implementation to comfortably modify them. It would take significant effort to analyze them first. I don't think the benefit is worth the effort.

**Comment by elfring** [comment]
>I don't think the benefit is worth the effort.

How much does this software depend on [the component “LEMON LALR(1) parser generator”](https://github.com/compiler-dept/lemon "Update candidate")?

-----

:thought_balloon: I suggest to avoid [ignorance of return values](https://cwe.mitre.org/data/definitions/252.html "Checking more return values") a bit more.
Would you like to detect every error situation as early as possible?

**Comment by pawelsalawa** [comment]
A lot. This is the parser used to understand SQL syntax of SQLite (for code assistant, code formatter, editing DDL objects, and a lot more). Why do you ask?

**Reply by elfring** [reply]
:thought_balloon: I became interested in [further collateral software evolution](https://www.sqlite.org/forum/forumpost/22a8cb4ef2 "Discussion: “Completion of error handling”").

**Comment by elfring** [comment]
:thought_balloon: I hope that more code reviewers and developers will become interested in corresponding software improvements.

**Comment by pawelsalawa** [comment]
I see. Just keep in mind, that SQLiteStudio uses relatively old version of Lemon parser and I had no need to update it so far. Updating is difficult, because I already introduced significant SQLiteStudio-specific changes to the Lemon included with SQLiteStudio.

**Reply by elfring** [reply]
>Updating is difficult, because I already introduced significant SQLiteStudio-specific changes to the Lemon included with SQLiteStudio.

* :thought_balloon: Can any of these parser adjustments become helpful also for a more recent software version?
* :crystal_ball: Would SQLite contributors become accordingly interested in further improvements?

**Comment by elfring** [comment]
:thinking: How do you think about to improve [static source code analysis](https://en.wikipedia.org/wiki/List_of_tools_for_static_code_analysis#C.2C_C.2B.2B "Tools for code analysis") also for this software?

**Comment by pawelsalawa** [comment]
I use embedded clang static analysis (as QtCreator plugin) since few months and I'm improving it on-the-fly whenever I touch files.

**Comment by elfring** [comment]
:thought_balloon: Would you become interested in [aspect-oriented software development](https://en.wikipedia.org/wiki/Aspect-oriented_programming "Documentation")?

**Comment by pawelsalawa** [comment]
Why all these questions?

**Comment by elfring** [comment]
:thought_balloon: Some advanced development tools can help more to fix remaining open issues.

:thinking: Can any more developers become aware that [error detection](https://en.wikipedia.org/wiki/Cross-cutting_concern#Examples "Documentation") belongs to known cross-cutting concerns?

**Reply by elfring** [reply]
:eyes: Some analysis tools can [point remaining open issues out](https://wiki.sei.cmu.edu/confluence/display/c/EXP12-C.+Do+not+ignore+values+returned+by+functions "Do not ignore values returned by functions").

:thought_balloon: I imagine that [pointcuts and advices can be developed accordingly](https://www.aspectc.org/ "Software development with AspectC++").
